.github/workflows/build_steps.yml

```diff
@@ -170,8 +170,8 @@ jobs:
       manylinux_image: ${{env.manylinux_image}}
       test_dirs: ${{env.test_dirs}}
 
-  python_tests:
-    if: inputs.job_type == 'follower'
+  python_tests_windows:
+    if: inputs.job_type == 'follower' && matrix.os == 'windows'
```

Comment:
I think it's more trouble than it's worth to split this job by OS because we will soon have a Mac build, and the copies will start to diverge in dangerous ways.

Reply:
There's no choice, the containerized service can only run on Linux

Reply:
Ah, that's trivial to work around. Because the `services` key interprets expressions, you can do something like

```
services: ${{fromJSON(matrix.test_services)}}
```

Where `test_services` can be `null` for Windows and `{"mongo": {"image": "...."}}` for Linux.

You have to double JSON-encoded the values and put them in those giant JSON blocks in `build.yml`

Reply:
I think it's rather confusing that each version's job group now has a weird skipped job in it:
![image](https://github.com/man-group/ArcticDB/assets/1662412/29f8bd27-1f52-484d-ad94-00d7ec785509)

Also, having minor differences in duplicated copies of code in an esoteric language is highly likely going to lead to maintenance issues and bugs. E.g. future maintainers will have to guess whether an inconsistency is intentional or due to carelessness. This is especially risky with tests as we can fool ourselves into believing something is being tested.

Reply:
Can I be a second opinion?

If they were otherwise identical I think the JSON hack is a great idea, but it looks like there's already a bunch of if statements? Like:

```
- name: Select Python (Linux)
```

```
- name: Windows Pagefile
```

```
- name: Collect crash dumps (Windows)
```

At this point it's just a question of 1 big if or many smaller ifs and I think it's probably much of a muchness as to which is "better".